### ISSUE TITLE
deprecate `fetch_atlas_aal`?

### ISSUE THREAD

**Issue by jeromedockes** [nilearn/nilearn#3086]
- It is not used in any example
- It's interface is inconsistent with other atlases, where the label for region `j` is found in `atlas.labels[atlas.indices[j]]` whereas other atlases have values 0, 1, ..., N in their map and label for region `j` is found in `atlas.labels[j]`

**Comment by jeromedockes** [comment]
# `fetch_atlas_allen_2011`

- only used in one plotting example (`01_plotting/plot_prob_atlas.py`) where it can easily be replaced by eg harvard-oxford
- the interface and documentation are confusing, eg the tuple of name, list of ints `rsn_indices` or the `networks` list of lists, see #3083

**Comment by bthirion** [comment]
I'm quite uncertain whether we should do it (these artlases are convenient for users, ans the cost is limited; but I understand your concerns regarding the lack of consistency). If a majority of people want to remove these atlases I'm fin with it.

**Comment by emdupre** [comment]
Are they already available in TemplateFlow ? I'd feel better deprecating if we can point to another python API for accessing them !

**Comment by jeromedockes** [comment]
> I'm quite uncertain whether we should do it (these artlases are convenient for users

Fair enough! It's just that as @NicolasGensollen worked on #3083 we realized their interface is cryptic and their documentation is incorrect; moreover the AAL one doesn't work (for some arguments) as reported in #3087. So if we have to deprecate some attributes to improve their interface, I wanted to see how people feel about getting rid of them altogether, esp. after the discussions in #3074 .

But I completely understand that if many users are relying on them, removing them is a big nuisance.

@emdupre  that's a good point! But IIUC [this templateflow issue](https://github.com/templateflow/templateflow/issues/101) it seems finding and accessing atlases in TemplateFlow is still a bit tricky?

**Comment by htwangtw** [comment]
> @emdupre that's a good point! But IIUC [this templateflow issue](https://github.com/templateflow/templateflow/issues/101) it seems finding and accessing atlases in TemplateFlow is still a bit tricky?

TemplateFlow doesn't have an extensive collection of atlases. It's actually quite difficult to add some atlases to TemplateFlow due to the lack of information on the exact MNI template used when generating the atlas. I had a quick look at both atlases and I could not find that information. 
I tried to check how much people have been using it based on search on all github repositories and code in python. (aal - 100 and allen - 31, comparing to craddock - 38; yeo - 60; basc - 63; harvard-oxford - 167, juelich - 6)
I am not sure if this provides any useful insight.

**Comment by Remi-Gau** [comment]
Update the following atlases are not used in any examples

- fetch_atlas_aal
- fetch_atlas_craddock_2012
- fetch_atlas_talairach
- fetch_atlas_schaefer_2018: is in templateFlow (see for example https://github.com/templateflow/tpl-MNI152NLin6Asym/tree/a68885d55e5669fd86b8c749ec97550d8fb5949d)

From @man-shu and I recent inventory of how nilearn is used, I could get the following counts for any function of that does some fetching:

https://nilearn.github.io/poia/notebooks/poia.html

| object                                |   count |
|:--------------------------------------|--------:|
| fetch_surf_fsaverage                  |     111 |
| fetch_atlas_harvard_oxford            |      97 |
| fetch_atlas_schaefer_2018             |      69 |
| fetch_atlas_msdl                      |      53 |
| fetch_haxby                           |      53 |
| fetch_atlas_aal                       |      50 |
| fetch_development_fmri                |      45 |
| fetch_atlas_basc_multiscale_2015      |      42 |
| fetch_abide_pcp                       |      37 |
| fetch_atlas_destrieux_2009            |      27 |
| fetch_adhd                            |      26 |
| fetch_atlas_yeo_2011                  |      26 |
| fetch_atlas_surf_destrieux            |      21 |
| fetch_icbm152_2009                    |      19 |
| fetch_coords_power_2011               |      17 |
| fetch_atlas_difumo                    |      15 |
| fetch_atlas_smith_2009                |      15 |
| fetch_oasis_vbm                       |      15 |
| fetch_neurovault_motor_task           |      14 |
| _fetch_files                          |      14 |
| _fetch_file                           |      10 |
| fetch_localizer_contrasts             |      10 |
| fetch_icbm152_brain_gm_mask           |       9 |
| fetch_atlas_juelich                   |       9 |
| fetch_atlas_pauli_2017                |       9 |
| fetch_atlas_talairach                 |       8 |
| fetch_atlas_craddock_2012             |       8 |
| fetch_spm_auditory                    |       7 |
| fetch_localizer_button_task           |       7 |
| fetch_neurovault_ids                  |       7 |
| fetch_surf_fsaverage5                 |       6 |
| fetch_coords_seitzman_2018            |       5 |
| fetch_miyawaki2008                    |       5 |
| fetch_neurovault                      |       5 |
| fetch_language_localizer_demo_dataset |       4 |
| fetch_coords_dosenbach_2010           |       4 |
| fetch_openneuro_dataset               |       3 |
| fetch_nyu_rest                        |       3 |
| fetch_cobre                           |       2 |
| fetch_fiac_first_level                |       2 |
| fetch                                 |       2 |
| fetch_atlas_allen_2011                |       2 |
| fetch_msdl_atlas                      |       2 |
| fetch_localizer_calculation_task      |       2 |
| fetch_ds000030_urls                   |       2 |
| fetch_grey_matter_mask                |       1 |
| fetch_localizer_first_level           |       1 |
| fetch_spm_multimodal_fmri             |       1 |
| fetch_files                           |       1 |
| fetch_surf_nki_enhanced               |       1 |
| fetch_openneuro_dataset_index         |       1 |
| fetch_harvard_oxford                  |       1 |
| fetch_haxby_simple                    |       1 |


given that `fetch_atlas_schaefer_2018` is in templateFlow and we do not use it in our example I would be in favor of starting a deprecation cycle to remove it with pointers on how to access it on templateFlow

**Comment by Remi-Gau** [comment]
Also the difumo atlas is available in templateFlow and only used in one of our examples (https://nilearn.github.io/stable/auto_examples/03_connectivity/plot_atlas_comparison.html).

I would be tempted to use another probabilistic atlas in this example (if possible) and to start a deprecation for the difumo atlas.

**Comment by Remi-Gau** [comment]
realizing that the same thing could be done for `fetch_atlas_harvard_oxford`.

**Comment by bthirion** [comment]
I am biased here, but I think that DiFuMo deserves a little push, since it's quite good for fMRI (better than Schaefer AFAICT), so I'd rather keep it if you don't mind.

**Comment by Remi-Gau** [comment]
I am ok either way.

BTW would it make sense to have an example that shows how to use templateFlow atlases with nilearn?

@htwangtw @emdupre

**Comment by htwangtw** [comment]
It's a tough one... 
Starting with atlas that's already on templateflow might be the most obvious next step technically, but I start to have other concerns after reading / seeing interaction of people from non neuroscience / fMRI background that wants to use the atlas.

Starting with deprecating atlas on templateflow will results in less well curated atlases stays in nilearn for longer. The atlas creator has either documented the template correctly or contributed to templateflow for the atlas to be in the repository. Nilearn would end up hosting atlases that didn't meet these criteria, which is a reasonable sign of quality IMHO. 

Having reviewed some papers from cs in the past year and seeing so many people using atlas that were 1) not examined on signal reconstruction 2) surface-based converting to volumetric use 3) poorly performed in the difumo benchmark 4) including cerebellum and subcortical structure but with no mention of how to handle missing coverage really got me thinking: how to we provide solutions that has already address these issue better without user jumping through extra hoops. 

I don't have a good strategy in mind, just putting the thoughts out here

**Comment by emdupre** [comment]
> BTW would it make sense to have an example that shows how to use templateFlow atlases with nilearn?

I think we should draft this s.t. it could go in the same release if we indeed end up deprecating some of the atlases we ship.

> Starting with deprecating atlas on templateflow will results in less well curated atlases stays in nilearn for longer. 

I'm not sure I understand this point, @htwangtw ? Your concern is that templateflow atlases are better documented than nilearn, or vice versa ?

**Comment by htwangtw** [comment]
> I'm not sure I understand this point, [@htwangtw](https://github.com/htwangtw) ? Your concern is that templateflow atlases are better documented than nilearn, or vice versa ?

I believe for all atlas released, they all have good amount of details that the original authors deemed important, and I think nilearn documentation does a good job on that front. However,for an atlas to be part of templateflow, one would need template information more than MNI152, but down to the versions.
